.github/workflows/dart.yml

```diff
@@ -82,7 +82,25 @@ jobs:
       - uses: ./
         with:
           sdk: ${{ matrix.sdk }}
-          architecture: ia32
+
+      - name: Run hello world
+        run: |
+          echo "main() { print('hello world'); }" > hello.dart
+          dart hello.dart
+      - run: dart --version
+
+  # Test using wildcard versions for the sdk parameter.
+  test_release_wildcards:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        sdk: [2.19.x, 3.1.x]
```

Comment:
Have you considered using `^2.19.0` instead? Then this would be more similar to how you specify SDK constraints, maybe we could even support more complicated SDK constraints in the future?

Reply:
Hmm, I haven't. We had a feature request for `2.19.x`, and I believe this may match the action support for one other language.

We could also consider just `2.19`; using just two version segments could be an implicit way to opt-into using the latest patch release for a version.

`^2.19.0` could also work. The current way this PR is implemented, we couldn't efficiently have an actual semver match - like getting the latest version that matches a version range (i.e., `'>=2.18.3 <2.19.2'`, ...). We could do that if we have a json method to call that either returned all the available sdk versions, or, all the versions that matched a given range. We're currently relying on a very specific usage of the google storage list API. It works the way we want it for this specific use case, but for anything more sophisticated we'd need to all the available sdk versions (perhaps available from a single well-known file in google storage - `https://storage.googleapis.com/dart-archive/channels/stable/release/versions.json`?).

Reply:
Up to you :)

Either option will work. Caret syntax feels more "Dart". Not sure if we ever need more complicated constraints (definitely wouldn't implement it now without a use case).

Reply:
The caret is actually kinda bad since ^3.1.0 means up to 4.0, which isn't the semantics we try to do here.

Reply:
That's true, caret means until the next major, so it doesn't quite cover the use case of latest patch version. So for that we'd need to support pub-like version ranges which seems inconvenient. So I think rather than making this too complicated we should look at other languages and make a "GitHub-y" solution.

Reply:
@sortie - good point about the caret syntax above.

I'd be fine w/ either `2.19` or `2.19.x`. They're both simple. If the two version variant (`2.19`) is more common for github actions then we should just do that. We _could_ support both patterns but I don't think the flexibility is really necessary here.

I'll re-work this PR to use `2.19`.